### DISCUSSION TITLE
Does DryIoc already work with dotnet 6/7 ?

### DISCUSSION THREAD

**Discussion by informatorius** [dadhi/DryIoc#570] [General]
My question is if DryIoc already work with dotnet 6/7 ?

**Comment by dadhi** [comment]
@informatorius What features of those platforms do you need?

- Current version v5.3.4 targets .NET Standard 2.0 - so it is compatible with .NET 6, 7
- The next version v5.4.0 will include the .NET 7 as a target, plus the next version of DryIoc.Microsoft.DependencyInjection will target 6 and 7 directly and will depend on [Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.Abstractions/#dependencies-body-tab)

**Comment by informatorius** [comment]
What is the difference between DryIoc and MS-DI? And what do I need DryIoc.Microsoft.DependencyInjection for?

**Comment by dadhi** [comment]
Here https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines

**Reply by xingzuhui** [reply]
Thanks for contributions to opensource.

**Reply by xingzuhui** [reply]
ms ：
Property injection
Injection based on name
Child containers
Custom lifetime management
Func<T> support for lazy initialization
Convention-based registration

But i need  injection base on attribute like  spring of java

**Reply by dadhi** [reply]
You may use DryIoc.MefAttributedModel package and get Export and Import attributes support, and many more. Just google what MEF (Microsoft Extensibily Framework) is.

**Comment by xingzuhui** [comment]
You Know，i am from china。Google and many other English websites access are hopes.
Yes,i am reading  blogs about MS.MEF . Thank you @dadhi .
By the way, DryIoc in Chinese means "Fuck IOC". And,dadhi,means "爸爸好"，“dad hello”。
However,DryIoc is helpful and effective.

**Reply by xingzuhui** [reply]
I can't delete my own issues。
